tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1802,51 +1804,85 @@ class LocationSet {
 public:
   /// Maximum number of indices supported
   const static uint32_t kMaxIndex = 2;
-  /// Maximum number of locations supported
-  // Typically we won't have that many stage input or output variables.
-  // Using 64 should be fine here.
-  const static uint32_t kMaxLoc = 64;
 
   LocationSet() {
     for (uint32_t i = 0; i < kMaxIndex; ++i) {
-      usedLocs[i].resize(kMaxLoc);
-      nextLoc[i] = 0;
+      // Default size. 64 should cover most cases without having to resize.
+      usedLocations[i].resize(64);
+      nextAvailableLocation[i] = 0;
     }
   }
 
-  /// Uses the given location.
+  /// Marks a given location as used.
   void useLoc(uint32_t loc, uint32_t index = 0) {
     assert(index < kMaxIndex);
-    usedLocs[index].set(loc);
-  }
 
-  /// Uses the next |count| available location.
-  int useNextLocs(uint32_t count, uint32_t index = 0) {
-    assert(index < kMaxIndex);
-    auto &locs = usedLocs[index];
-    auto &next = nextLoc[index];
-    while (locs[next])
-      next++;
+    auto &set = usedLocations[index];
+    if (loc <= set.size()) {
```

Comment:
```suggestion
    if (loc >= set.size()) {
```

Reply:
Ah yes, thanks. I'd like to unit test this TBH...